### PR TITLE
Fix exe being rebuilt every time on windows

### PR DESCRIPTION
**Pull request by parched**
Fixes https://github.com/pyinstaller/pyinstaller/issues/2239

### CODE REVIEW

File: PyInstaller/utils/win32/winmanifest.py [hunk 1]
```diff
@@ -1085,7 +1085,7 @@ def create_manifest(filename, manifest, console, uac_admin=False, uac_uiaccess=F
     # only write a new manifest if it is different from the old
     need_new = not os.path.exists(filename)
     if not need_new:
-        old_xml = ManifestFromXMLFile(filename).toprettyxml()
+        old_xml = ManifestFromXMLFile(filename).toprettyxml().replace('\r','')
```
Comment (parched) [inline PyInstaller/utils/win32/winmanifest.py:1088]:
```suggestion
        old_xml = ManifestFromXMLFile(filename).toprettyxml().replace('\r', '')
```

File: PyInstaller/utils/win32/winmanifest.py [hunk 2]
```diff
@@ -1085,7 +1085,7 @@ def create_manifest(filename, manifest, console, uac_admin=False, uac_uiaccess=F
...
         new_xml = manifest.toprettyxml().replace('\r','')
```
Comment (parched) [inline PyInstaller/utils/win32/winmanifest.py:1089]:
```suggestion
        new_xml = manifest.toprettyxml().replace('\r', '')
```

File: PyInstaller/building/api.py [hunk 3]
```diff
@@ -476,6 +476,13 @@ def __init__(self, *args, **kwargs):
             self.toc.append(("pyi-disable-windowed-traceback", "", "OPTION"))
 
         if is_win:
+            if not self.icon:
+                # --icon not specified; use default from bootloader folder
+                self.icon = os.path.join(
+                    os.path.dirname(os.path.dirname(__file__)),
+                    'bootloader', 'images',
+                    'icon-console.ico' if self.console else 'icon-windowed.ico')
+
```
Comment (parched) [inline PyInstaller/building/api.py:485]:
```suggestion
            if not self.icon:
                # --icon not specified; use default from bootloader folder
                if self.console:
                    icon = 'icon-console.ico'
                else:
                    icon = 'icon-windowed.ico'
                self.icon = os.path.join(
                    os.path.dirname(os.path.dirname(__file__)),
                    'bootloader', 'images', icon)
```

**Review by parched — COMMENTED** [review]
Lint fixes

**Comment by rokm** [comment]
Nicely spotted! 

Please add [a news entry](https://pyinstaller.readthedocs.io/en/stable/development/changelog-entries.html), then this is good to go.

**Comment by parched** [comment]
> Nicely spotted!
> 
> Please add [a news entry](https://pyinstaller.readthedocs.io/en/stable/development/changelog-entries.html), then this is good to go.

Ok, done.

**Pull request merged**